server-function/src/index.ts

```diff
@@ -0,0 +1,57 @@
+import path from 'path';
+import fs from 'mz/fs';
+import mimeTypes from 'mime-types';
+
+const UP_PATH_REGEXP = /(?:^|[\\/])\.\.(?:[\\/]|$)/;
+
+export async function isFile(filePath: string): Promise<boolean> {
+  try {
+    const stats = await fs.stat(filePath);
+    return stats.isFile();
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      return false;
+    }
+    throw err;
+  }
+}
+
+export class Server {
+  constructor(private directory: string, private extensions: string[] = ['.html']) {
+  }
+  public async handle(url: string): Promise<any> {
```

Comment:
why any?
it's a pretty rigid type (action, sometimes status, sometimes fullPath and contentType)